zeebe/atomix/cluster/src/main/java/io/atomix/raft/roles/LeaderAppender.java

```diff
@@ -467,8 +475,14 @@ private void handleInstallResponseFailure(
       final RaftMemberContext member, final InstallRequest request, final Throwable error) {
     // Reset the member's snapshot index and offset to resend the snapshot from the start
     // once a connection to the member is re-established.
-    member.setNextSnapshotIndex(0);
-    member.setNextSnapshotChunk(null);
+    final boolean isTimeout =
+        error instanceof TimeoutException
+            || (error != null && error.getCause() instanceof TimeoutException);
```

Comment:
:question: What were the case where it would be the cause? Do you think it could also be more deeply nested?

Reply:
`error` is usually `CompletionException` so we always have to get the cause. I don't think it can be more deeply nested.